### PR TITLE
Initial templating

### PR DESCRIPTION
**Pull request by PaulJWright**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        - linux: py311-devdeps
+    secrets:
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+
+  publish_pure:
+        needs: [test, docs]
+        uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish_pure_python.yml@main
+        with:
+          python-version: '3.10'
+          test_extras: 'tests'
```
Comment (samaloney) [inline .github/workflows/ci.yml]: The dependancy is called test in the project.toml
```suggestion
          test_extras: 'test'
```

File: .cruft.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  "context": {
+    "cookiecutter": {
+      "package_name": "MagIEpy",
+      "module_name": "magiepy",
+      "short_description": "Python utilities for the Magnetometer Network of Ireland (MagIE)",
+      "author_name": "DIAS Solar",
+      "author_email": "",
+      "project_url": "magie.ie",
+      "license": "BSD 3-Clause",
+      "minimum_python_version": "3.9",
```
Comment (samaloney) [inline .cruft.json]: I think just at end of py3.10 support might want to move to 3.10 (https://scientific-python.org/specs/spec-0000/)

File: .readthedocs.yaml
Comment (samaloney) [inline .readthedocs.yaml:1]: have you created an RTD for this?

File: .ruff.toml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+select = ["E", "F", "W", "UP", "PT"]
+extend-ignore = [
+  # pycodestyle (E, W)
+  "E501", # LineTooLong                             # TODO! fix
+  # pytest (PT)
+  "PT001", # Always use pytest.fixture()
+  "PT004", # Fixtures which don't return anything should have leading _
+  "PT007", # Parametrize should be lists of tuples  # TODO! fix
+  "PT011", # Too broad exception assert             # TODO! fix
+  "PT023", # Always use () on pytest decorators
```
Comment (samaloney) [inline .ruff.toml]: Some of these could be removed as don't apply ...

**Review by samaloney — APPROVED** [review]
LGTM - few comment taken or leave them and a fix for the purewheeels.

File: .readthedocs.yaml
Comment (PaulJWright) [reply .readthedocs.yaml:1]: I have not, I guess this is something you might have to set up, does it need user details etc?

File: .cruft.json (hunk 2 above)
Comment (PaulJWright) [reply .cruft.json]: Yes, that's probably a good plan.

Comment (PaulJWright) [reply .cruft.json]: I have attempted to change everything to 3.10, if you could give a quick sweep that would be appreciated

File: .ruff.toml (hunk 3 above)
Comment (PaulJWright) [reply .ruff.toml]: Do you have an opinion?

Comment (samaloney) [inline .ruff.toml]: Maybe this just remove the one that were marked to be fixed?
```suggestion
  # pytest (PT)
  "PT001", # Always use pytest.fixture()
  "PT004", # Fixtures which don't return anything should have leading _
  "PT023", # Always use () on pytest decorators
```

**Review by samaloney — APPROVED** [review]

File: .readthedocs.yaml
Comment (samaloney) [reply .readthedocs.yaml:1]: Yea I requested a group account ages ago but not go so I've created them all under my account should really add @drsophiemurray to them all too.

**Pull request merged**